petab/v1/parameters.py

```diff
@@ -332,9 +332,8 @@ def append_overrides(overrides):
         if not model.has_entity_with_id(p):
             parameter_ids[p] = None
 
-    # remove parameters that occur in the condition table and are overridden
-    #  for ALL conditions
-    for p in condition_df.columns[~condition_df.isnull().any()]:
+    # parameters that are overridden via the condition table are not allowed
+    for p in condition_df.columns:
```

Comment:
Instead of try-except, could do one of these, but fine as is
```python
for p in set(condition_df.columns).intersection(parameter_ids):
    del parameter_ids[p]

for p in condition_df.columns:
    if p in parameter_ids:
        del parameter_ids[p]
```

Reply:
Right, but then one has to look up each element twice.